security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -721,29 +722,57 @@ func concatCerts(certsPEM []string) []byte {
 	return certChain.Bytes()
 }
 
-func (sc *SecretManagerClient) getConfigTrustBundle() []byte {
-	sc.configTrustBundleMutex.RLock()
-	defer sc.configTrustBundleMutex.RUnlock()
-	return sc.configTrustBundle
-}
+// UpdateConfigTrustBundle : Update the Configured Trust Bundle in the secret Manager client
+func (sc *SecretManagerClient) UpdateConfigTrustBundle(caCerts []string) error {
+	var update bool = false
+
+	newTbSet := map[string]struct{}{}
+	for _, cert := range caCerts {
+		newTbSet[cert] = struct{}{}
+	}
 
-func (sc *SecretManagerClient) setConfigTrustBundle(trustBundle []byte) {
 	sc.configTrustBundleMutex.Lock()
 	defer sc.configTrustBundleMutex.Unlock()
-	sc.configTrustBundle = trustBundle
-}
 
-// UpdateConfigTrustBundle : Update the Configured Trust Bundle in the secret Manager client
-func (sc *SecretManagerClient) UpdateConfigTrustBundle(trustBundle []byte) error {
-	existingBundle := sc.getConfigTrustBundle()
-	if bytes.Equal(existingBundle, trustBundle) {
+	for cert := range newTbSet {
+		if _, ok := sc.configTrustBundle[cert]; !ok {
+			update = true
+		}
+	}
+	for cert := range sc.configTrustBundle {
+		if _, ok := newTbSet[cert]; !ok {
+			update = true
+		}
+	}
+	// Do not trigger updates if there is no diff
+	if !update {
 		return nil
 	}
-	sc.setConfigTrustBundle(trustBundle)
+	sc.configTrustBundle = newTbSet
 	sc.CallUpdateCallback(security.RootCertReqResourceName)
 	return nil
 }
 
 func (sc *SecretManagerClient) mergeConfigTrustBundle(rootCert []byte) []byte {
-	return pkiutil.AppendCertByte(sc.getConfigTrustBundle(), rootCert)
+	duplicate := false
+	certs := []string{}
+	rootCertBytes := []byte{}
+
+	sc.configTrustBundleMutex.RLock()
+	if len(rootCert) != 0 {
+		_, duplicate = sc.configTrustBundle[string(rootCert)]
```

Comment:
Seeing the places mergeConfigTrustBundle() is called, it seems rootCert could contain multiple certs, right?